src/main/java/org/junit/runners/Parameterized.java

```diff
@@ -365,7 +396,13 @@ private TestWithParameters createTestWithNotNormalizedParameters(
         @SuppressWarnings("unchecked")
         private static List<Object> allParameters(
                 TestClass testClass, FrameworkMethod parametersMethod) throws Throwable {
-            Object parameters = parametersMethod.invokeExplosively(null);
+            Object parameters;
+            try {
+                parameters = parametersMethod.invokeExplosively(null);
+            } catch (AssumptionViolatedException e) {
```

Comment:
I've implemented the first part. 

`ErrorReportingRunner` is already applied in `RunnerBuilder.safeRunnerForClass()`, seems like we should not care about other exceptions.